### DISCUSSION TITLE
SSR, cache-control & getInitialProps on Vercel + NextJS

### DISCUSSION THREAD

**Discussion by ItsWendell** [vercel/next.js#14167] [Help]
Dear NextJS Community,

Server-side rending is important in some of the pages we have in our platform, I'm using Apollo Client and a version of a `withApollo` (https://github.com/vercel/next.js/issues/9542) H.O.C. to achieve SSR on the pages I need it, this HOC uses `getInitialProps`.

Now for some pages this might be a bit slow, so I started thinking about caching. I figured out that hosting on Vercel, I can set `cache-control` headers for serverless functions, and I was wondering if this is the right way to go if I put for example:

```
ctx.res?.setHeader(
        "cache-control",
        "s-maxage=7200, stale-while-revalidate"
      );
```

in my pages `getInitialProps`, also read a bit about SSR with incremental builds, but I'm not sure if that's reasonable to do in this case, since we might have 50.000+ of these pages at one point and they need to be updated when new information is available. There is only a few limited actions that we need to do though before this information needs to be re-regenerated.

Any tips / advice?

**Comment by mikestopcontinues** [comment]
Hey, did you ever find a solution for this? I'm looking around since we still don't have getStaticProps in _app.

**Comment by ItsWendell** [comment]
@mikestopcontinues I basically doing the exact thing I described. We changed the way we use that higher order component for Apollo and use the method described in the current examples but with getServersideProps. We do use vercel's CDN cache (enabled by passing those headers described) for essential SEO pages and our search pages.

**Reply by mikestopcontinues** [reply]
Thanks for the info! I'm just about ready to deploy today, so I'll give it a spin. I was concerned with this method because I've got to cache by domain, but I guess more than likely Vercel caches by domain and not deployment ID or whatnot.
